extension/scripts/global/functions/dom.js

```diff
@@ -0,0 +1,365 @@
+					break;
+			}
+		}
+	} else if (header.find("i.fa-caret-down")) {
+		header.find("i.fa-caret-down").classList.add("fa-caret-up");
+		header.find("i.fa-caret-down").classList.remove("fa-caret-down");
+
+		order = "asc";
+	} else if (header.find("i.fa-caret-up") || order === "desc") {
+		header.find("i.fa-caret-up").classList.add("fa-caret-down");
+		header.find("i.fa-caret-up").classList.remove("fa-caret-up");
+
+		order = "desc";
+	} else {
+		header.appendChild(document.newElement({ type: "i", class: "fas fa-caret-up" }));
+
+		order = "asc";
+	}
+	for (let h of table.findAll("th")) {
+		if (h === header) continue;
+
+		if (h.find("i")) h.find("i").remove();
+	}
+
+	let rows;
+	if (!table.find("tr:not(.heading)")) rows = [];
+	else {
+		rows = [...table.findAll("tr:not(.header)")];
+		rows = sortRows(rows);
+	}
+
+	for (let row of rows) table.appendChild(row);
+
+	function sortRows(rows) {
+		if (order === "asc") {
+			rows.sort((a, b) => {
+				const helper = sortHelper(a, b);
+
+				return helper.a - helper.b;
+			});
+		} else if (order === "desc") {
+			rows.sort((a, b) => {
+				const helper = sortHelper(a, b);
+
+				return helper.b - helper.a;
+			});
+		}
+
+		return rows;
+
+		function sortHelper(elementA, elementB) {
+			elementA = elementA.find(`*:nth-child(${columnPlace})`);
+			elementB = elementB.find(`*:nth-child(${columnPlace})`);
+
+			let valueA, valueB;
+			if (elementA.hasAttribute("sort-type")) {
+				switch (elementA.getAttribute("sort-type")) {
+					case "date":
+						valueA = elementA.getAttribute("value");
+						valueB = elementB.getAttribute("value");
+
+						if (Date.parse(valueA)) valueA = Date.parse(valueA);
+						if (Date.parse(valueB)) valueA = Date.parse(valueB);
+						break;
+					case "css-dataset":
+						valueA =
+							elementA.dataset[
+								getComputedStyle(elementA)
+									.getPropertyValue("--currentValue")
+									.match(/attr\(data-(.*)\)/i)[1]
+							];
+						valueB =
+							elementB.dataset[
+								getComputedStyle(elementB)
+									.getPropertyValue("--currentValue")
+									.match(/attr\(data-(.*)\)/i)[1]
+							];
+						break;
+					default:
+						console.warn("Attempting to sort by a non-existing type.", elementA.getAttribute("sort-type"));
+						return;
+				}
+			} else if (elementA.hasAttribute("value")) {
+				valueA = elementA.getAttribute("value");
+				valueB = elementB.getAttribute("value");
+			} else {
+				valueA = elementA.innerText;
+				valueB = elementB.innerText;
+			}
+
+			let a, b;
+			if (isNaN(parseFloat(valueA))) {
+				if (valueA.includes("$")) {
+					a = parseFloat(valueA.replace("$", "").replace(/,/g, ""));
+					b = parseFloat(valueB.replace("$", "").replace(/,/g, ""));
+				} else {
+					a = valueA.toLowerCase().localeCompare(valueB.toLowerCase());
+					b = 0;
+				}
+			} else {
+				a = parseFloat(valueA);
+				b = parseFloat(valueB);
+			}
+
+			return { a, b };
+		}
+	}
+}
+
+function showLoadingPlaceholder(element, show) {
```

Comment:
Not sure if this should be in dom.js file, from what I understand these should all be utility functions like newElement and friends, but this one stands out as being more like it should be part of a feature/page

Reply:
For now it would indeed make more sense for it to be present only in `popup.js`, but it will be used more later on:
- Any page that has stat estimates (profile, faction, userlist, etc.).
- Any page that shows user data (profile, faction and company).
- Importing and exporting settings to the remote server.

Almost all functions inside of these could also fit into `utilities.js`, but `dom.js` is specifically for functions that require the dom/window.

Reply:
I see what you mean, a generic loading overlay/temporary widget to be reused, makes sense